### PR TITLE
Create an empty registry if external registry is not connected

### PR DESCRIPTION
**Pull request by DeividasK**
Paired with @cedric-cordenier

### CODE REVIEW

File: core/capabilities/registry.go [hunk 1]
```diff
@@ -199,3 +200,30 @@ func NewRegistry(lggr logger.Logger) *Registry {
 		lggr: lggr.Named("CapabilitiesRegistry"),
 	}
 }
+
+// TestMetadataRegistry is a test implementation of the metadataRegistry
```
Comment (bolekk) [inline core/capabilities/registry.go:204]: nit: I don't think it should be called "Test", maybe "Dummy" or "Default" or something similiar

File: go.mod [hunk 2]
```diff
@@ -102,15 +102,15 @@ require (
...
 	go.uber.org/multierr v1.11.0
 	go.uber.org/zap v1.27.0
-	golang.org/x/crypto v0.25.0
-	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56
-	golang.org/x/mod v0.19.0
-	golang.org/x/net v0.27.0
-	golang.org/x/sync v0.7.0
-	golang.org/x/term v0.22.0
-	golang.org/x/text v0.16.0
+	golang.org/x/crypto v0.26.0
```
Comment (bolekk) [inline go.mod:105]: are all these expected?

File: core/capabilities/registry.go [hunk 3]
```diff
@@ -199,3 +200,30 @@ func NewRegistry(lggr logger.Logger) *Registry {
...
+
+func (t *TestMetadataRegistry) LocalNode(ctx context.Context) (capabilities.Node, error) {
+	peerID := p2ptypes.PeerID{}
+	workflowDON := capabilities.DON{
+		ID:            1,
+		ConfigVersion: 1,
+		Members: []p2ptypes.PeerID{
+			peerID,
+		},
+		F:                0,
```
Comment (bolekk) [inline core/capabilities/registry.go:216]: I hope this won't cause any validation issues down the line. Are we ever going to look at these values in other componenets?

**Review by bolekk — DISMISSED** [review]

File: core/capabilities/registry.go (hunk 1 above)
Comment (DeividasK) [reply core/capabilities/registry.go:204]: I agree with your sentiment. I had this discussion with @cedric-cordenier, too. Given that I am using it for testing at the moment and we don't have a clear product need, `Test` seems "calling it what it is".

File: go.mod (hunk 2 above)
Comment (DeividasK) [reply go.mod:105]: This is what showed up after `go get github.com/smartcontractkit/chainlink-common@<hash> && make gomodtidy`, so I assume so 🤷‍♂️

Comment (DeividasK) [reply go.mod:105]: Ah, these are from Ryan bumping the deps for the generators (in common)

File: core/capabilities/registry.go (hunk 3 above)
Comment (DeividasK) [reply core/capabilities/registry.go:216]: > Are we ever going to look at these values in other components?

Not planned atm.

**Review by cedric-cordenier — DISMISSED** [review]

**Review by archseer — DISMISSED** [review]

**Comment by lukaszcl** [comment]
@DeividasK you have to update the branch and merge develop. Otherwise PR will be kicked out of merge queue due to recent changes to CI on develop.

File: integration-tests/go.mod [hunk 4]
```diff
@@ -33,8 +33,8 @@ require (
 	github.com/slack-go/slack v0.12.2
 	github.com/smartcontractkit/chain-selectors v1.0.21
 	github.com/smartcontractkit/chainlink-automation v1.0.4
-	github.com/smartcontractkit/chainlink-common v0.2.2-0.20240816204408-654165b6ee33
-	github.com/smartcontractkit/chainlink-testing-framework v1.34.5
```
Comment (lukaszcl) [inline integration-tests/go.mod:37]: @DeividasK please use `chainlink-testing-framework v1.34.5`. I'm not sure about `chainlink-common v0.2.2-0.20240816204408-654165b6ee33`. Do you have any reasons to use a different version for it? Also, once you update the mods in `integration-tests/` then please do:
```
cd integration-tests/ 
go mod tidy
```
and
```
cd integration-tests/load
go mod tidy
```

**Review by lukaszcl — CHANGES_REQUESTED** [review]

File: integration-tests/go.mod (hunk 4 above)
Comment (DeividasK) [reply integration-tests/go.mod:37]: > I'm not sure about chainlink-common v0.2.2-0.20240816204408-654165b6ee33. Do you have any reasons to use a different version for it?

Yes, the latest version has the changes that fix keystone integration tests.

**Review by lukaszcl — APPROVED** [review]

**Review by ettec — APPROVED** [review]

File: core/capabilities/registry.go (hunk 3 above)
Comment (archseer) [reply core/capabilities/registry.go:216]: Ah it would be nice if this was configurable: I need `f = 1` for e2e testing

**Pull request merged**